Tests/BackendIntegrationTests/StoreKitObserverModeIntegrationTests.swift

```diff
@@ -81,6 +81,33 @@ class StoreKit2ObserverModeIntegrationTests: StoreKit1ObserverModeIntegrationTes
 
 }
 
+class StoreKit2ObserverModeDisabledIntegrationTests: StoreKit1ObserverModeIntegrationTests {
+
+    override class var storeKitVersion: StoreKitVersion { .storeKit2 }
+
+    override class var observerMode: Bool { false }
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+    }
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testObservingTransactionThrowsIfObserverModeNotEnabled() async throws {
```

Comment:
We'll also need an integration test to verify that the transaction is posted, etc.

Reply:
I think that should be covered by `testObservingTransactionUnlocksEntitlement`?